### PR TITLE
init infra cluster controller

### PR DESCRIPTION
**Pull request by zirain**
Signed-off-by: hejianpeng <hejianpeng2@huawei.com>

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
xref #145

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/controllers/infra/cluster_controller.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+		return errors.Wrapf(err, "failed to delete CAPI Cluster %s/%s", infraCluster.Namespace, infraCluster.Name)
+	}
+
+	return nil
+}
+
+func (r *ClusterController) reconcile(ctx context.Context, infraCluster *infrav1.Cluster) (ctrl.Result, error) {
+	switch infraCluster.Spec.InfraType {
+	case infrav1.AWSClusterInfraType:
+		panic("not implemented")
```
Comment (hzxuzhonghu) [inline pkg/controllers/infra/cluster_controller.go:218]: ?? This is the target work, right

Comment (zirain) [reply pkg/controllers/infra/cluster_controller.go]: mark this as TODO, will implement it later.

File: manifests/charts/cluster-operator/templates/rbac.yaml [hunk 2]
```diff
@@ -3,6 +3,19 @@ kind: ClusterRole
 metadata:
   name: kurator-cluster-operator
 rules:
+# cluster operaotr crd
+- apiGroups:
+  - infra.kurator.dev
+  resources:
+  - '*'
+  verbs:
+  - create
```
Comment (hzxuzhonghu) [inline manifests/charts/cluster-operator/templates/rbac.yaml:12]: rm

File: manifests/charts/cluster-operator/templates/rbac.yaml [hunk 3]
```diff
@@ -3,6 +3,19 @@ kind: ClusterRole
...
+  - delete
```
Comment (hzxuzhonghu) [inline manifests/charts/cluster-operator/templates/rbac.yaml:13]: rm

File: pkg/apis/infra/v1alpha1/common_types.go [hunk 4]
```diff
@@ -14,12 +14,11 @@ See the License for the specific language governing permissions and
...
-package scheme
+package v1alpha1
 
-import (
-	clusterv1alpha1 "kurator.dev/kurator/pkg/apis/cluster/v1alpha1"
+const (
+	// ClusterAnnotation is the annotation that is added to a machine object to skip IAM profile reconciliation.
+	SkipIAMProfileAnnotation = "kurator.dev/skip-iam-profile"
+	// ClusterFinalizer allows ClusterController to clean up associated resources before removing it from apiserver.
+	ClusterFinalizer = "cluster.infra.kurator.dev"
```
Comment (hzxuzhonghu) [inline pkg/apis/infra/v1alpha1/common_types.go:23]: move to controller

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**